hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -0,0 +1,722 @@
+                      doc="Controls the PyTorch Lightning flag 'find_unused_parameters' for the DDP plugin. "
+                          "Setting it to True comes with a performance hit.")
+    pl_limit_train_batches: Optional[int] = \
+        param.Integer(default=None,
+                      doc="PyTorch Lightning trainer flag 'limit_train_batches': Limit the training dataset to the "
+                          "given number of batches.")
+    pl_limit_val_batches: Optional[int] = \
+        param.Integer(default=None,
+                      doc="PyTorch Lightning trainer flag 'limit_val_batches': Limit the validation dataset to the "
+                          "given number of batches.")
+    pl_profiler: Optional[str] = \
+        param.String(default=None,
+                     doc="The value to use for the 'profiler' argument for the Lightning trainer. "
+                         "Set to either 'simple', 'advanced', or 'pytorch'")
+    monitor_gpu: bool = param.Boolean(default=False,
+                                      doc="If True, add the GPUStatsMonitor callback to the Lightning trainer object. "
+                                          "This will write GPU utilization metrics every 50 batches by default.")
+    monitor_loading: bool = param.Boolean(default=True,
+                                          doc="If True, add the BatchTimeCallback callback to the Lightning trainer "
+                                              "object. This will monitor how long individual batches take to load.")
+
+    @property
+    def use_gpu(self) -> bool:
+        """
+        Returns True if a GPU is available, and the self.max_num_gpus flag allows it to be used. Returns False
+        otherwise (i.e., if there is no GPU available, or self.max_num_gpus==0)
+        """
+        if self.max_num_gpus == 0:
+            return False
+        from health_ml.utils.common_utils import is_gpu_available
+        return is_gpu_available()
+
+    def num_gpus_per_node(self) -> int:
+        """
+        Computes the number of gpus to use for each node: either the number of gpus available on the device
+        or restrict it to max_num_gpu, whichever is smaller. Returns 0 if running on a CPU device.
+        """
+        import torch
+        available_gpus = torch.cuda.device_count()  # type: ignore
+        num_gpus = available_gpus if self.use_gpu else 0
+        message_suffix = "" if self.use_gpu else ", but not using them because use_gpu == False"
+        logging.info(f"Number of available GPUs: {available_gpus}{message_suffix}")
+        if 0 <= self.max_num_gpus < num_gpus:
+            num_gpus = self.max_num_gpus
+            logging.info(f"Restricting the number of GPUs to {num_gpus}")
+        elif self.max_num_gpus > num_gpus:
+            logging.warning(f"You requested max_num_gpus {self.max_num_gpus} but there are only {num_gpus} available.")
+        return num_gpus
+
+
+class DeepLearningConfig(WorkflowParams,
```

Comment:
I think this whole class here is obsolete. It contains stuff that should mostly be handled by the lightning container itself.